appengine/myria_web_main.py

```diff
@@ -199,21 +199,24 @@ def get_connection_string(self):
         return connection_string
 
     def base_template_vars(self):
-        try:
-            url = "http://169.254.169.254/latest/meta-data/public-hostname"
-            result = urlfetch.fetch(url).content
-        except:
-            result = "localhost"
-
         if self.app.ssl:
             uri_scheme = "https"
         else:
             uri_scheme = "http"
+
+        try:
+            meta = "http://169.254.169.254/latest/meta-data/public-hostname"
+            hostname = urlfetch.fetch(meta).content
+            jupyterURL = "{s}://{h}:{p}".format(
+                s=uri_scheme, h=hostname, p="8888")
```

Comment:
We only have the REST_PORT listed here, do we have an environment variable
for the Jupyter notebook port that myria-cluster works with or should I
just introduce a new variable here (MYRIA_JUPYTER_PORT?)

On Tue, Jul 26, 2016 at 9:46 PM, Tobin Baker <notifications@github.com>
wrote:

> In appengine/myria_web_main.py
> <https://github.com/uwescience/myria-web/pull/310#discussion_r72380267>:
>
> >          if self.app.ssl:
> >              uri_scheme = "https"
> >          else:
> >              uri_scheme = "http"
> > +
> > +        try:
> > +            meta = "http://169.254.169.254/latest/meta-data/public-hostname"
> > +            hostname = urlfetch.fetch(meta).content
> > +            jupyterURL = "{s}://{h}:{p}".format(
> > +                s=uri_scheme, h=hostname, p="8888")
>
> I think we should be pulling the jupyter host/port from env vars in
> app.yaml, like the MyriaX host/port:
>
> env_variables:
>   MYRIAX_REST_HOST: localhost
>   MYRIAX_REST_PORT: 8753
>
> That makes it simple to configure with myria-cluster.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/uwescience/myria-web/pull/310/files/a75d2153e7703373d6884fb7f652823b705c35d9..063747b972853278ae91563dc88696d00e612ed2#r72380267>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADFus0l31XTk9e_1uzLq8a31JcOobIwfks5qZuKUgaJpZM4JQWhO>
> .
>